packages/astro/test/test-utils.js

```diff
@@ -171,6 +177,17 @@ export async function loadFixture(inlineConfig) {
 		config,
 		resolveUrl,
 		fetch: async (url, init) => {
+			if (config.vite?.server?.https) {
+				init = {
+					dispatcher: new Agent({
+						connect: {
+							rejectUnauthorized: false,
+						},
+						allowH2: true,
+					}),
```

Comment:
What's this for? I have no knowledge of this

Reply:
This tells undici fetch to use a custom dispatcher. This lets you customise some of the HTTP options for the request. `rejectUnauthorized` lets you ignore invalid server certificates, which is needed here as the dev server will always use self-signed certs. `allowH2` enables HTTP/2 requests in fetch, if the server supports it.

Reply:
I'll add comments for both of these